lib/membrane/pipeline.ex

```diff
@@ -67,50 +63,58 @@ defmodule Membrane.Pipeline do
   require Membrane.Core.Message, as: Message
 
   @typedoc """
-  Defines options that can be passed to `start/3` / `start_link/3` and received
-  in `c:handle_init/2` callback.
+  Defines options passed to the `start/3` and `start_link/3` and subsequently received
+  in the `c:handle_init/2` callback.
   """
   @type pipeline_options :: any
 
+  @typedoc "The Pipeline name"
   @type name :: GenServer.name()
 
+  @typedoc "List of configurations used by `start/3` and `start_link/3`."
   @type config :: [config_entry()]
+
+  @typedoc "Defines configuration value used by the `start/3` and `start_link/3`."
   @type config_entry :: {:name, name()}
 
+  @typedoc """
+  Defines the return value of the `start/3` and `start_link/3`."
+  """
   @type on_start ::
           {:ok, supervisor_pid :: pid, pipeline_pid :: pid}
           | {:error, {:already_started, pid()} | term()}
 
+  @typedoc """
+  The Pipeline state.
```

Comment:
Just for the sake of consistency 
```suggestion
  The pipeline state.
```

Reply:
I believe it is also applicable to a few other places